src/driven/worker/storage/keyvalue.rs

```diff
@@ -0,0 +1,129 @@
+use std::sync::Arc;
+
+use sqlx::{postgres::PgRow, FromRow, Postgres, Row};
+
+use crate::domain::{
+    error::Error,
+    worker::{KeyValue, WorkerKeyValueDrivenStorage},
+    Result,
+};
+
+use super::PostgresStorage;
+
+pub struct PostgresWorkerKeyValueDrivenStorage {
+    storage: Arc<PostgresStorage>,
+}
+impl PostgresWorkerKeyValueDrivenStorage {
+    pub fn new(storage: Arc<PostgresStorage>) -> Self {
+        Self { storage }
+    }
+}
+
+#[async_trait::async_trait]
+impl WorkerKeyValueDrivenStorage for PostgresWorkerKeyValueDrivenStorage {
+    async fn find(
+        &self,
+        worker_id: &str,
+        key: Option<String>,
+        page: &u32,
+        page_size: &u32,
+    ) -> Result<Vec<KeyValue>> {
+        let offset = (page_size * (page - 1)) as i64;
+        let page_size = *page_size as i64;
+
+        let mut query = String::from(
+            r#"
+            SELECT 
+                kv.worker, 
+                kv."key", 
+                kv.value,
+                kv.type,
+                kv.secure
+            FROM
+                kv
+            WHERE kv.worker = $1
+        "#,
+        );
+
+        if key.is_some() {
+            query.push_str(" AND kv.\"key\" = $4");
+        }
```

Comment:
Ideally we should use `ILIKE '%xx%'` allowing partial search on keys. For example, if user want's to get all the `postgres_xx` keys, they can put key as `postgres` and the ILIKE will return all the keys which match with `%postgres%`